src/bms/parse/rng.rs

```diff
@@ -1,22 +1,104 @@
 //! Random generator for parsing BMS format.
 //!
 //! [`RngMock`] can be used for testing the parser result with some random scopes.
+//! [`RandRng`] can be used for generating random numbers with [`rand`] crate.
+//!
+//! [`rand`]: https://crates.io/crates/rand
 
-use std::ops::RangeInclusive;
+use num::{BigUint, iter::RangeInclusive};
 
 /// A random generator for parsing BMS.
 pub trait Rng {
     /// Generates a random integer within the `range`. Returning the number outside the range will result weird.
-    fn generate(&mut self, range: RangeInclusive<u32>) -> u32;
+    ///
+    /// - Example:
+    /// ```rust
+    /// use bms_rs::bms::parse::rng::{Rng, RngMock};
+    /// use num::BigUint;
+    ///
+    /// let mut rng = RngMock([BigUint::from(1u64)]);
+    /// let n = rng.generate(num::range_inclusive(BigUint::from(1u64), BigUint::from(10u64)));
+    /// assert!(n >= BigUint::from(1u64) && n <= BigUint::from(10u64));
+    /// ```
+    fn generate(&mut self, range: RangeInclusive<BigUint>) -> BigUint;
```

Comment:
[`RangeInclusive`](https://doc.rust-lang.org/std/ops/struct.RangeInclusive.html) I wrote means std's one , that is a superset of num's `RangeInclusive`.